x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/logging/Throttler.java

```diff
@@ -9,150 +9,205 @@
+     * @param loggingInterval the frequency to run a task to emit repeated log messages
+     */
+    public Throttler(Throttler oldThrottler, TimeValue loggingInterval) {
+        this(loggingInterval, oldThrottler.clock, oldThrottler.threadPool, new ConcurrentHashMap<>(oldThrottler.logExecutors));
     }
 
     /**
      * This should only be used directly for testing.
      */
-    Throttler(
-        TimeValue resetInterval,
-        TimeValue durationToWait,
-        Clock clock,
-        ThreadPool threadPool,
-        ConcurrentMap<String, LogExecutor> logExecutors
-    ) {
-        Objects.requireNonNull(durationToWait);
-        Objects.requireNonNull(threadPool);
-
-        this.resetInterval = Objects.requireNonNull(resetInterval);
-        this.durationToWait = Duration.ofMillis(durationToWait.millis());
+    Throttler(TimeValue loggingInterval, Clock clock, ThreadPool threadPool, ConcurrentMap<String, LogExecutor> logExecutors) {
+        this.threadPool = Objects.requireNonNull(threadPool);
+        this.loggingInterval = Objects.requireNonNull(loggingInterval);
         this.clock = Objects.requireNonNull(clock);
         this.logExecutors = Objects.requireNonNull(logExecutors);
+    }
+
+    public void init() {
+        cancellableTask.set(startRepeatingLogEmitter());
+    }
+
+    private Scheduler.Cancellable startRepeatingLogEmitter() {
+        classLogger.debug(() -> Strings.format("Scheduling repeating log emitter with interval [%s]", loggingInterval));
 
-        this.cancellableTask.set(startResetTask(threadPool));
+        return threadPool.scheduleWithFixedDelay(this::emitRepeatedLogs, loggingInterval, threadPool.executor(UTILITY_THREAD_POOL_NAME));
     }
 
-    private Scheduler.Cancellable startResetTask(ThreadPool threadPool) {
-        classLogger.debug(() -> format("Reset task scheduled with interval [%s]", resetInterval));
+    private void emitRepeatedLogs() {
+        if (isRunning.get() == false) {
+            return;
+        }
+
+        final ReentrantReadWriteLock.WriteLock writeLock = lock.writeLock();
+
+        writeLock.lock();
+        try {
+            for (var iter = logExecutors.values().iterator(); iter.hasNext();) {
+                var executor = iter.next();
+                executor.logRepeatedMessages();
+                iter.remove();
+            }
+        } finally {
+            writeLock.unlock();
+        }
+    }
 
-        return threadPool.scheduleWithFixedDelay(logExecutors::clear, resetInterval, threadPool.executor(UTILITY_THREAD_POOL_NAME));
+    public void warn(Logger logger, String message) {
+        execute(message, logger::warn);
     }
 
-    public void setDurationToWait(TimeValue durationToWait) {
-        this.durationToWait = Duration.ofMillis(durationToWait.millis());
+    public void warn(Logger logger, String message, Throwable t) {
+        execute(message, enrichedMessage -> logger.warn(enrichedMessage, t));
```

Comment:
I moved the lambda in here to attempt to encapsulate the functionality since we are calling an alien method from a synchronized block and I'm trying to reduce the likelihood of a deadlock in the future.